packages/jest-haste-map/src/createJestWorkerWorkerFactory.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type {WorkerFactory, WorkerInterface} from './types';
+
+export default function createJestWorkerWorkerFactory({
```

Comment:
Getting some Java vibes from this name. I wonder if there's a way to make it more clear what's going on here.

Reply:
I merged the factory and worker and renamed the concept to `MetadataExtractor`. I hope that helps.